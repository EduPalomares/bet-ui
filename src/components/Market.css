.market {
	padding:15px;
	border-top: 3px solid #79879c;
}

.market .objective {
	color: #7f8da1;
	margin-bottom: 10px;
}

.market .selection {
	width:auto;
	display: inline-block; 
	vertical-align:top;
	margin-right: 10px;
	padding:10px 15px;
	border:2px solid #79879c;
	border-radius:10px;
	text-align:center;
	font-size: 11px;
	cursor: pointer;
	transition: all .35s;
}

.market .selection.selected {
	background:#07b037;
}

.market .selection:hover {
	background:#DDD;
	transition: all .35s;
}