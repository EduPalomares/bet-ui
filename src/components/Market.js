import React, { Component } from 'react';
import Selection from './Selection';
import './Market.css';

export default class Market extends Component {
	render() {

		const selections = this.props.selections.map((selection) =>
			{
				return <Selection key={selection.id} selection={selection} />
			}
		);

		return (
			<div className="market">
				<div className="objective">{this.props.marketname}</div>
				{selections}
			</div>
		);
	}
}