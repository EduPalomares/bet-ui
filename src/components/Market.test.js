import React from 'react';
import Market from './Market';

const props = {
	selections: []
}

const wrapper = shallow(<Market {...props} />);

test('render match snapshot', () => {
	expect(wrapper).toMatchSnapshot();
});

test('render objective', () => {
	expect(wrapper.find('.objective').length).toBe(1);
});

