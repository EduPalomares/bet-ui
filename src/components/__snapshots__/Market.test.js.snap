// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Market
    selections={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="objective"
/>,
        Array [],
      ],
      "className": "market",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "objective",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="objective"
/>,
          Array [],
        ],
        "className": "market",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "objective",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
